lib/iolog/log_writer.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iolog
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+	"time"
+)
+
+const (
+	// MaxEntrySizeBytes is the maximum allowable size of a log entry
+	// before it will be broken into multiple entries
+	MaxEntrySizeBytes = 16 * 1024
```

Comment:
It's actually pretty huge, but it was determined by the number of leftover bits in the final two bytes of the header. We can make it smaller if we like, but reasonable (ie: not 64B) smaller values will leave us with some unused bits in the header (probably not a bad thing if you consider those optional future flags). I could make it 8K or 4K and save 1 or 2 bits respectively for future use. I'm down to tweak it to optimize for latency, or take suggestions as I have no strong opinion one way or the other since almost all entries right now will be well below 16K!